### PR TITLE
Feat: Added Count, Patch and Delete method to client

### PR DESCRIPTION
**Pull request by MIA-Deltat1995**

### CODE REVIEW

File: testhelper/helpers.go [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
-import "github.com/mia-platform/go-crud-service-client/internal/types"
+import (
+	"encoding/json"
+
+	"github.com/mia-platform/go-crud-service-client/internal/types"
+)
 
 type Filter types.Filter
+
+func ParseResponseToNdjson[TResource any](response []TResource) (string, error) {
```
Comment (MIA-Deltat1995) [inline testhelper/helpers.go:11]: Is it possible to import this function in other packages?

File: CHANGELOG [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# CHANGELOG
```
Comment (davidebianchi) [inline CHANGELOG:1]: Change log is not needed, in this project we use the releases

File: client.go [hunk 3]
```diff
@@ -138,11 +177,9 @@ func (c Client[Resource]) PatchById(ctx context.Context, id string, body PatchBo
 	return resource, nil
 }
 
-// List get the resources of the collection with the specified filter. It is limited by default
-// and with a max page of 200 elements (by default).
-// If you want to take more elements, use pagination
-func (c Client[Resource]) List(ctx context.Context, options Options) ([]Resource, error) {
-	req, err := c.client.NewRequestWithContext(ctx, http.MethodGet, "", nil)
+// Patch updates resources in bulk using commands in PatchBody
+func (c Client[Resource]) Patch(ctx context.Context, body PatchBody, options Options) (*Resource, error) {
```
Comment (davidebianchi) [inline client.go:181]: From [this](https://github.com/mia-platform/go-crud-service-client/pull/5#pullrequestreview-1462037191) we should call this method PatchMany

File: client.go [hunk 4]
```diff
@@ -198,3 +235,20 @@ func (c Client[Resource]) DeleteById(ctx context.Context, id string, options Opt
 	}
 	return nil
 }
+
+// Delete allow to remove multiple resources.
+func (c Client[Resource]) Delete(ctx context.Context, options Options) error {
```
Comment (davidebianchi) [inline client.go:240]: Also this, maybe DeleteMany it's better?

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: client.go (hunk 3 above)
Comment (fredmaggiowski) [reply client.go:181]: I agree (with myself)

File: client.go (hunk 4 above)
Comment (fredmaggiowski) [reply client.go:240]: yep

File: testhelper/helpers.go (hunk 1 above)
Comment (fredmaggiowski) [inline testhelper/helpers.go:11]: please add *testing.T as first argument and call t.Helper() in the fn

Comment (fredmaggiowski) [reply testhelper/helpers.go:11]: yes it can be imported by third party packages as `testhelper.ParseResponseToNdjson(...`

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -12,12 +12,15 @@ and we want a way to interact to it with a standard client.
...
 
-- GetById: `GET /:id`
-- PatchById: `PATCH /:id`
-- Export: `GET /export`
+- GetById: `GET`
 - List: `GET /`
+- Count: `GET /count`
+- Export: `GET /export`
+- PatchById: `PATCH /:id`
+- Patch: `PATCH /`
```
Comment (fredmaggiowski) [inline README.md:20]:
```suggestion
- PatchMany: `PATCH /`
```

File: README.md [hunk 6]
```diff
@@ -12,12 +12,15 @@ and we want a way to interact to it with a standard client.
...
 - Create: `POST /`
 - DeleteById: `DELETE /:id`
+- Delete: `DELETE /`
```
Comment (fredmaggiowski) [inline README.md:23]:
```suggestion
- DeleteMany: `DELETE /`
```

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

**Review by fredmaggiowski — APPROVED** [review]

**Pull request merged**